topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -473,6 +475,33 @@ public void setTopicCount(int topics) {
                 action.execute().onComplete(actionResult -> {
                     LOGGER.debug("{}: Executing handler for action {} on topic {}", logContext, action, lockName);
                     action.result = actionResult;
+
+                    String keytag = namespace + ":" + "KafkaTopic" + "/" + key.asKubeName().toString();
+                    Optional<Meter> metric = metrics.meterRegistry().getMeters()
+                            .stream()
+                            .filter(meter -> meter.getId().getName().equals(METRICS_PREFIX + "resource.state") &&
+                                    meter.getId().getTags().contains(Tag.of("kind", "KafkaTopic")) &&
+                                    meter.getId().getTags().contains(Tag.of("name", key.asKubeName().toString())) &&
+                                    meter.getId().getTags().contains(Tag.of("resource-namespace", namespace))
+                            ).findFirst();
+                    if (metric.isPresent()) {
+                        // remove metric so it can be re-added with new tags
+                        metrics.meterRegistry().remove(metric.get().getId());
+                        LOGGER.debug("Removed metric " + METRICS_PREFIX + "resource.state{}", keytag);
```

Comment:
Why are you mixing `{}` and `+` string concatenation for logging? Using this should be easier:
`LOGGER.debug("Removed metric {}.resource.state{}", METRICS_PREFIX, keytag);`